### PR TITLE
Make FFmpeg log level configurable

### PR DESCRIPTION
**Pull request by mthrok**
and set the default level to 8 for less verbose logging.

Undesired logs are one of the loudest UX complains we get.
Yet, loading media files involves uncertainty which is
difficult to debug without debug log.

This commit introduces utility functions to configure logging level
so that we can ask users to enable it when they encounter an issue,
while defaulting to non-verbose option.

### CODE REVIEW

File: torchaudio/utils/ffmpeg_utils.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return torch.ops.torchaudio.ffmpeg_get_log_level()
+
+
+def set_log_level(level: int):
+    """Set the log level of FFmpeg (libavformat etc)
+
+    Arguments:
+        level (int): Log level. The larger, the more verbose.
+
+            The following values are valid values, the corresponding ``ffmpeg``'s
```
Comment (carolineechen) [inline torchaudio/utils/ffmpeg_utils.py:18]: what happens if they enter an invalid value? should we add a check?

**Review by carolineechen — COMMENTED** [review]
utility function looks good, just want to double check default log value -- description mentions 0 but it is 8 in the code. generally enabling it for errors (16) by default makes more sense to me, but do you find this to be too noisy?

> we can ask users to enable it when they encounter an issue
is there a good place we could add this message upon failure so users are aware of this log setting utility

File: torchaudio/utils/ffmpeg_utils.py (hunk 1 above)
Comment (mthrok) [reply torchaudio/utils/ffmpeg_utils.py:18]: I think they will boil down to a simple level comparison, so it's safe to put other values.

https://ffmpeg.org/doxygen/4.1/log_8c_source.html#l00316

It's just normally, log-levels are set by `ffmpeg` CLI, which exposes the listed values. Let me rephrase, `common values`, instead of saying `valid values`.

**Comment by mthrok** [comment]
> description mentions 0 but it is 8 in the code. 

Description also says 8. In test code it's reset to 0. 

> generally enabling it for errors (16) by default makes more sense to me, but do you find this to be too noisy?

16 can repeat messages across decoding units, so, decoding one audio file can still flood the log.

> is there a good place we could add this message upon failure so users are aware of this log setting utility

Decoder errors are not necessarily failure, that's why it's tricky. You would still get the tensor and keep going. But if you are training a model with many files, then the size of the training log could blow up with the decoding issues that is not an issue for DL practitioners.

**Review by carolineechen — APPROVED** [review]
>Description also says 8.

I meant it says 0 in the PR description, but yea code looks good if 8 is intended.

For the second point, was more wondering if there was a good spot in the ffmpeg documentation or logging output to let users know that this logging functionality exists (this is not blocking)

File: test/torchaudio_unittest/utils/ffmpeg_utils_test.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from torchaudio.utils import ffmpeg_utils
+from torchaudio_unittest.common_utils import skipIfNoFFmpeg, PytorchTestCase
+
+
+@skipIfNoFFmpeg
+class TestFFmpegUtils(PytorchTestCase):
+    """Smoke test for ffmpeg_utils moduls"""
```
Comment (hwangjeff) [inline test/torchaudio_unittest/utils/ffmpeg_utils_test.py:7]:
```suggestion
    """Smoke test for ffmpeg_utils module"""
```

File: test/torchaudio_unittest/common_utils/case_utils.py [hunk 3]
```diff
@@ -112,22 +113,8 @@ class TorchaudioTestCase(TestBaseMixin, PytorchTestCase):
...
-    global _IS_FFMPEG_AVAILABLE
-    if _IS_FFMPEG_AVAILABLE is None:
-        try:
-            from torchaudio.io import StreamReader  # noqa: F401
-
-            _IS_FFMPEG_AVAILABLE = True
-        except Exception:
-            _IS_FFMPEG_AVAILABLE = False
-    return _IS_FFMPEG_AVAILABLE
+    return torchaudio._extension._FFMPEG_INITIALIZED
```
Comment (hwangjeff) [inline test/torchaudio_unittest/common_utils/case_utils.py:117]: does this change relate to log level configurability?

**Review by hwangjeff — COMMENTED** [review]
have you considered introducing constants for the different log levels, e.g. `FATAL` and `WARNING`, similar to `logging`?

File: test/torchaudio_unittest/common_utils/case_utils.py (hunk 3 above)
Comment (mthrok) [reply test/torchaudio_unittest/common_utils/case_utils.py:117]: not technically. but as I write the test, I realized that the logic is duplicated in library and test util.

**Comment by mthrok** [comment]
> I meant it says 0 in the PR description, but yea code looks good if 8 is intended.
> 

Oh my bad, fixed the description. 

> For the second point, was more wondering if there was a good spot in the ffmpeg documentation or logging output to let users know that this logging functionality exists (this is not blocking)

Good point, let me update the doc.

**Comment by mthrok** [comment]
> have you considered introducing constants for the different log levels, e.g. `FATAL` and `WARNING`, similar to `logging`?

I thought about it, but unlike the `logging` module which is all about logging, this utility module is things more general about ffmpeg runtime configuration, so, I did not think putting alias at module level makes much sense, then thought adding some ENUM-like features for the `set_log_level` is kinda overkill.

**Comment by mthrok** [comment]
@carolineechen let me revert the default log level modification, as it might be disruptive if other libraries / existing code base expects log level with already set.

**Review by hwangjeff — APPROVED** [review]
